src/overlays/actors/ovl_En_Diving_Game/z_en_diving_game.c

```diff
@@ -26,70 +52,532 @@ const ActorInit En_Diving_Game_InitVars = {
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 素もぐりＧＯ ☆☆☆☆☆ \n" VT_RST);
+    this->actor.room = -1;
+    this->actor.scale.x = 0.01f;
+    this->actor.scale.y = 0.012999999f;
+    this->actor.scale.z = 0.0139999995f;
+    if (D_809EF0B0 != 0) {
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ もういてる原 ☆☆☆☆☆ \n" VT_RST);
+        this->unk_31F = 1;
+        Actor_Kill(&this->actor);
+    } else {
+        D_809EF0B0 = 1;
+        this->actor.targetMode = 0;
+        this->actor.colChkInfo.mass = 0xFF;
+        this->actionFunc = func_809EDCB0;
+    }
+}
+
+void EnDivingGame_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDivingGame* this = THIS;
+    
+    if (this->unk_31F == 0) {
+        gSaveContext.timer1State = 0;
+    }
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_809ED9E0(EnDivingGame* this, GlobalContext* globalCtx) {
+    EnExRuppy* attached;
+    Vec3f rupeePos;
+
+    rupeePos.x = (Rand_ZeroOne() - 0.5f) * 30.0f + this->actor.world.pos.x;
+    rupeePos.y = (Rand_ZeroOne() - 0.5f) * 20.0f + (this->actor.world.pos.y + 30.0f);
+    rupeePos.z = (Rand_ZeroOne() - 0.5f) * 20.0f + this->actor.world.pos.z;
+    attached = (EnExRuppy*)Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_EX_RUPPY,
+                                              rupeePos.x, rupeePos.y, rupeePos.z, 0,
+                                              (s16)Rand_CenteredFloat(3500.0f) - 1000, this->unk_2A6, 0);
+    if (attached != NULL) {
+        attached->actor.speedXZ = 12.0f;
+        attached->actor.velocity.y = 6.0f;
+    }
+}
+
+s32 func_809EDB08(EnDivingGame* this, GlobalContext* globalCtx) {
+    s32 var;
+
+    if ((gSaveContext.timer1State == 0xA) && (Gameplay_InCsMode(globalCtx) == 0)) {
+        gSaveContext.timer1State = 0;
+        func_800F5B58();
+        func_80078884(NA_SE_SY_FOUND);
+        this->actor.textId = 0x71AD; // Too bad! Practice hard and come back!
+        func_8010B680(globalCtx, this->actor.textId, NULL);
+        this->unk_2A4 = 0;
+        this->unk_292 = 5;
+        this->unk_31E = this->unk_2A8 = this->unk_29C = this->unk_2A2 = this->unk_2A4;
+        func_8002DF54(globalCtx, NULL, 8);
+        this->actionFunc = func_809EE048;
+        return 1;
+    } else {
+        var = 5;
+        if ((gSaveContext.eventChkInf[3] & 0x100) != 0) {
+            var = 10;
+        }
+        if (this->unk_2A4 >= var) {
+            gSaveContext.timer1State = 0;
+            this->unk_2A4 = 0;
+            this->unk_2A2 = this->unk_2A4;
+            this->unk_29C = this->unk_2A4;
+            this->unk_2A8 = this->unk_2A4;
+            this->unk_31E = this->unk_2A4;
+            if ((gSaveContext.eventChkInf[3] & 0x100) == 0) {
+                this->actor.textId = 0x4055; // "Hey! Congratulations! I've got something very nice for you! ..."
+            } else {
+                this->actor.textId = 0x405D; // "Hey! Congratulations!!"
```

Comment:
i know this wasnt your doing, but can you remove any text box comments. we used to do this a while back but have since stopped